### DISCUSSION TITLE
build improvements

### DISCUSSION THREAD

**Discussion by andreas-arkulpa** [coollabsio/coolify#393] [Feature Requests]
Hello @andrasbacsai,

is there any option considered to stop a running build/deployment or automatically cancel the build when there is a new commit? That would be great to have!

We had a case where the deployment/build of a node project took about 45.000 seconds (normally it is about 600-1000 seconds) according to the coolify-dashboard. Afterwards everything crashed (needed to restart the server and coolify to get it running again) and it seems like the build was kind of hanging according to the build-logs.
Have you got any idea what this could have caused?

Yours
Andreas

**Comment by andrasbacsai** [comment]
Hey, 

Yes. I've added to the next version, coming today (or latest, tomorrow). 

I've been searching for a long time for a queue system (I'm using BullMQ) that has a feature to cancel jobs, but I could not find any. But today morning, I was thinking and found another way around, not canceling the queue but stopping the docker container, which will cause the queue job to be canceled.

About why it's building that long, that is a good question. Most probably some resources hanging (network / CPU / memory / IO). I need to debug that on the server to know why it's hanging.

**Reply by andreas-arkulpa** [reply]
Thanks a lot! I will try that as soon as the update is available.
Have you got any Idea what could cause the crash as mentioned above?

**Reply by andrasbacsai** [reply]
Not sure tbh. As the logs are already lost (I think), it's hard to tell.


I've released the new version that has the feature of canceling new running builds.

**Reply by andreas-arkulpa** [reply]
thanks a lot!

**Comment by nowebpublicidade** [comment]
Hi, everyone!

I'm a recent Coolify user, I'm currently testing some features and, in my tests, can't auto-deploy from a github commit. I connected Coolify Application with my github profile, and it's reading all archives from the gh repo, but can't deploy from any commit. Is there any specific configuration on my gh profile or something I need to do? 

Can someone help me?

Thanks!

**Reply by andreas-arkulpa** [reply]
Hi @nowebpublicidade,
have you enabled commit / pull-request previews in the coolify application?

Maybe you need to enabled that there as well.

**Comment by Dhruvgera** [comment]
Hi! Is there any way in the latest version to deploy only latest commits? Would be really helpful!
